vscode-extension/src/aiConfigEditor.ts

```diff
@@ -144,23 +144,58 @@ export class AIConfigEditorProvider implements vscode.CustomTextEditorProvider {
             return;
           }
 
+          // TODO: Should we skip events with no content changes?
+          // if (e.contentChanges.length === 0) {
+          //   console.log(
+          //     `changeDocumentSubscription ${e.document.uri} - skipping event because there are no content changes`
+          //   );
+          //   return;
+          // }
+
           // TODO: saqadri - instead of sending the entire document to the webview,
           // can ask it to reload the document from the server
 
           // Update webview immediately, and server state should be updated in the background.
+          console.log(
+            `changeDocumentSubscription ${e.document.uri} -- updating webview`
+          );
           updateWebview();
 
           // Notify server of updated document
           if (editorServer) {
-            console.log("changeDocumentSubscription -- updating server");
-
             // TODO: saqadri - decide if we want to await here or just fire and forget
-            await initializeServerState(editorServer.url, e.document);
-          }
+            try {
+              console.log("changeDocumentSubscription -- updating server");
+              await updateServerState(editorServer.url, e.document);
+            } catch (e) {
+              if (isWebviewDisposed || !editorServer) {

```

Comment:
What are the use cases where this is true because of the 2nd condition instead of the 1st (ie: `isWebviewDisposed` is false, but `editorServer` is null)? Is it when we first try to open the file in AIConfig editor before it's had time to initialize the server connection?

Reply:
Hmm, that's a good point. In theory I guess that scenario may be able to happen. I had just added this check for completeness but maybe we do actually want to surface the error if this scenario *does* happen. I'll remove the `!editorServer` check